src/Structs.vala

```diff
@@ -20,4 +20,75 @@ namespace FeedReader {
         string data;
     }
 
+    private struct ResourceMetadata
+	{
+		private const string CACHE_GROUP = "cache";
+		private const string ETAG_KEY = "etag";
+		private const string LAST_MODIFIED_KEY = "last_modified";
+
+		string? etag;
+		string? last_modified;
+
+		public ResourceMetadata()
+		{
+		}
+
+		public ResourceMetadata.from_file(string filename)
+		{
+			var config = new KeyFile();
+			try
+			{
+				config.load_from_file(filename, KeyFileFlags.NONE);
+				try { this.etag = config.get_string(CACHE_GROUP, ETAG_KEY); }
+				catch (KeyFileError.KEY_NOT_FOUND e) {}
+				catch (KeyFileError.GROUP_NOT_FOUND e) {}
+				try { this.last_modified = config.get_string(CACHE_GROUP, LAST_MODIFIED_KEY); }
+				catch (KeyFileError.KEY_NOT_FOUND e) {}
+				catch (KeyFileError.GROUP_NOT_FOUND e) {}
+			}
+			catch (KeyFileError e)
+			{
+				Logger.warning(@"FaviconMetadata.from_file: Failed to load $filename: " + e.message);
+			}
+			catch (FileError e)
+			{
+				Logger.warning(@"FaviconMetadata.from_file: Failed to load $filename: " + e.message);
+			}
+		}
+
+		public void save_to_file(string filename)
+		{
+			if(this.etag == null && this.last_modified == null)
+			{
+				if(FileUtils.unlink(filename) != 0)
+					Logger.warning(@"FaviconMetadata.save_to_file: Error deleting metadata file $filename");
+			}
+			else
+			{
+				var config = new KeyFile();
+				config.set_string(CACHE_GROUP, ETAG_KEY, this.etag);
+				config.set_string(CACHE_GROUP, LAST_MODIFIED_KEY, this.last_modified);
+				try
+				{
+					config.save_to_file(filename);
+				}
+				catch (FileError e)
+				{
+					Logger.warning(@"FaviconMetadata.save_to_file: Failed to save metadata file $filename: " + e.message);
+				}
+			}
+		}
+
+        public GLib.DateTime? lastModifiedDateTime()
+        {
+            if(last_modified == null)
+                return null;
+
+            GLib.Time time = GLib.Time();
+            time.strptime(last_modified, "%a, %d %b %Y %H:%M:%S %Z");
+
+            return new GLib.DateTime.local(1900 + time.year, 1 + time.month, time.day, time.hour, time.minute, time.second);
```

Comment:
> Years since 1900

https://valadoc.org/glib-2.0/GLib.Time.year.html

> Months since January – [0, 11]

https://valadoc.org/glib-2.0/GLib.Time.month.html